app/Http/Controllers/SignUpController.php

```diff
@@ -23,19 +23,17 @@ public function signUp(Request $request)
         if ($password1 != $password2) {
             return view('signup', ['errorMessage' => 'Passwords do not match.']);
         } else {
-            $users = DB::table('users')->where('email', $email)->get();
-            if (!empty($users)) {
+            $users = User::where('email', $email)->get();
+            if (sizeof($users) > 0) {
                 return view('signup', ['errorMessage' => 'That email is already in use']);
             } else {
-                $id = DB::table('users')->insertGetId(
-                    [
-                        'email' => $email, 
-                        'password' => bcrypt($password1),
-                        'name' => $email,
-                        'totpSecret' => GoogleAuthenticator::generateRandom()
-                    ]
-                );
-                if (Auth::loginUsingId($id)) {
+                $user = new User;
+                $user->email = $email;
+                $user->password = bcrypt($password1);
+                $user->name = $email;
+                $user->totpSecret = GoogleAuthenticator::generateRandom();
+                $user->save();
```

Comment:
Assigning each property to the user is alright, but remember you could also use `User::create` with all the parameters you define in the `fillable` property of the User model. No need to change this.